### DISCUSSION TITLE
How can I get data from column using enableRowSelection?

### DISCUSSION THREAD

**Discussion by virusek392** [KevinVandy/material-react-table#453] [General]
like the question in the topic.
Best regards,
Lukasz

**Comment by KevinVandy** [comment]
There is an entire docs page about this https://www.material-react-table.com/docs/guides/row-selection

**Reply by virusek392** [reply]
Ok. But i dont understand how display date from column ID or Data
![obraz](https://user-images.githubusercontent.com/107401577/229433140-c9a3000a-fd85-4691-adc4-dbf1a89755ba.png)
if use rowSelection then i have this:
![obraz](https://user-images.githubusercontent.com/107401577/229433360-9506b874-d1ef-439f-bcd5-9a7d09ba3b92.png)
this my code:
```
 useEffect(() => {

    //do something when the row selection changes...

    console.info({ rowSelection });

  }, [rowSelection]);

.....
      <MaterialReactTable
      columns={columns}
      data={dane}
      getRowId={(row) => row.userId} 
      enableTopToolbar={false} //hide top toolbar
      state={{ isLoading: ladowanie, rowSelection  }}
    
      enableFullScreenToggle={false}
      onRowSelectionChange={setrowSelection} //connect internal row selection state to your own

     // onRowSelectionChange={setRowSelection}
     muiTableBodyRowProps={({ row }) => ({

      onClick: row.getToggleSelectedHandler(),

      sx: { cursor: 'pointer' },

    })}
      enableMultiRowSelection={false}
 
    />
```

**Reply by KevinVandy** [reply]
`table.getSelectedRowModel().rows` if you want to read all of the row data

**Reply by virusek392** [reply]
OK, can you write me simple example by using table.getSelectedRowModel().rows?. I did use it, byt it's dosen;t work :(
 muiTableBodyRowProps={({ row,table }) => ({
      
      onClick: row.getToggleSelectedHandler(),

      onClick: (row,table) => {
        row.getToggleSelectedHandler();
       console.log( table.getSelectedRowModel().rows);
      },
